### PR TITLE
Package refactoring

### PR DESCRIPTION
**Pull request by GiBg1aN**
- package `ru.mystamps.web.model` renamed to `ru.mystamps.web.controller.dto`
- package `ru.mystamps.web.validation.jsr303` renamed to `ru.mystamps.web.support.beanvalidation`

Addressed to #591

### CODE REVIEW

**Comment by php-coder** [comment]
@GiBg1aN Hi! Thank you for your contribution!

Could you split this into 2 commits (to have each renaming will be in the separate commit)? And also update the commits description to not mention the original issue (and we won't have spam comments in the issue from the temporary commits).

**Comment by php-coder** [comment]
Also don't forget to update here:
```
src/main/config/checkstyle-suppressions.xml:    <suppress checks="HideUtilityClassConstructor" files="ru.mystamps.web.model" />
```

**Comment by php-coder** [comment]
There are also translations for the validators that should be updated as well: see `src/main/resources/ru/mystamps/i18n/ValidationMessages.properties` and `src/main/resources/ru/mystamps/i18n/ValidationMessages_ru.properties`

**Comment by GiBg1aN** [comment]
@php-coder thank you for answering! I just splitted it in two commits and properly distributed the 3 missing files to them.

File: src/main/resources/ru/mystamps/i18n/ValidationMessages_ru.properties [hunk 1]
```diff
@@ -1,50 +1,50 @@
-javax.validation.constraints.NotNull.message = Поле обязательно для заполнения
```
Comment (php-coder) [inline src/main/resources/ru/mystamps/i18n/ValidationMessages_ru.properties]: Could you modify this file without fully decoding it?

**Comment by php-coder** [comment]
@GiBg1aN Please, look at the errors from our bot and try to fix them.

File: src/main/resources/ru/mystamps/i18n/ValidationMessages_ru.properties (hunk 1 above)
Comment (GiBg1aN) [inline src/main/resources/ru/mystamps/i18n/ValidationMessages_ru.properties]: Ok, I'm gonna try the replacement with another editor

Comment (php-coder) [inline src/main/resources/ru/mystamps/i18n/ValidationMessages_ru.properties]: If you have `sed` then you can do this with the command: `sed -i 's|validation\.jsr303|support.beanvalidation|' src/main/resources/ru/mystamps/i18n/ValidationMessages.properties`

**Comment by php-coder** [comment]
> Couldn’t find pom.log. sortpom-maven-plugin result is unknown

This and other similar warnings just have been fixed in the commit 10949526aa7db5f554faf0a70c120247d078f518 They will gone if you rebase your branch against the latest master. But it's not required.

**Comment by php-coder** [comment]
@GiBg1aN I see that you use your nickname as an author for the commits. Generally I advise to use a real name (but it's not required of course). So, consider updating the author value in the commits (`git commit user.name "John Dow" && git commit --am --reset-author`).

File: src/main/java/ru/mystamps/web/support/beanvalidation/ReleaseDateIsNotInFuture.java [hunk 2]
```diff
@@ -36,7 +36,8 @@
 @Constraint(validatedBy = ReleaseDateIsNotInFutureValidator.class)
 @Documented
 public @interface ReleaseDateIsNotInFuture {
-	String message() default "{ru.mystamps.web.validation.jsr303.ReleaseDateIsNotInFuture.message}";
+	String message() default "{ru.mystamps.web.support.beanvalidation"
+		+ ".ReleaseDateIsNotInFuture.message}";
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/support/beanvalidation/ReleaseDateIsNotInFuture.java]: Why do we need this? If you were trying to fix CheckStyle complains about long line, then in this particular case, it's better to ignore them with a special comment:
```java
// CheckStyle: ignore LineLength for next 1 line
```

File: src/main/java/ru/mystamps/web/controller/CategoryController.java [hunk 3]
```diff
@@ -42,10 +42,10 @@
 import ru.mystamps.web.Url;
 import ru.mystamps.web.controller.converter.annotation.Category;
 import ru.mystamps.web.controller.converter.annotation.CurrentUser;
+import ru.mystamps.web.controller.dto.AddCategoryForm;
 import ru.mystamps.web.controller.editor.ReplaceRepeatingSpacesEditor;
 import ru.mystamps.web.dao.dto.LinkEntityDto;
 import ru.mystamps.web.dao.dto.SeriesInfoDto;
-import ru.mystamps.web.controller.dto.AddCategoryForm;
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/controller/CategoryController.java]: The changes like that is irrelevant to the validation package renaming and should be part of the previous commit instead. I'm trying to have the commits that are "atomic" and modifies only one particular thing at time.

Could you move such changes from this commit to the previous (it requires some diff/patch/git rebase magic)?

File: src/main/java/ru/mystamps/web/support/beanvalidation/ExistingActivationKey.java [hunk 4]
```diff
@@ -34,7 +34,8 @@
 @Constraint(validatedBy = ExistingActivationKeyValidator.class)
 @Documented
 public @interface ExistingActivationKey {
-	String message() default "{ru.mystamps.web.validation.jsr303.ExistingActivationKey.message}";
+	String message() default "{ru.mystamps.web.support.beanvalidation"
+		+ ".ExistingActivationKey.message}";
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/support/beanvalidation/ExistingActivationKey.java]: This should be a single line.

**Comment by php-coder** [comment]
Thank you again!
Overall it looks good but some changes are required to make it mergeable.

**Comment by php-coder** [comment]
Could you also remove this lines as they not needed anymore: https://github.com/php-coder/mystamps/blob/10949526aa7db5f554faf0a70c120247d078f518/src/main/config/findbugs-filter.xml#L11-L19 (they should go to the appropriate commit).

**Comment by GiBg1aN** [comment]
- [x] merged master commits
- [x] file editing grouped in the proper commit
- [x] removed unused code from `findbugs-filter.xml`
- [x] longline warnings suppressed

I hope that now everything is ok

File: src/main/java/ru/mystamps/web/controller/dto/ActivateAccountForm.java [hunk 5]
```diff
@@ -27,11 +27,12 @@
...
+import ru.mystamps.web.support.beanvalidation.ExistingActivationKey;
+import ru.mystamps.web.support.beanvalidation.FieldsMatch;
+import ru.mystamps.web.support.beanvalidation.FieldsMismatch;
+import ru.mystamps.web.support.beanvalidation.UniqueLogin;
 import ru.mystamps.web.validation.ValidationRules;
-import ru.mystamps.web.validation.jsr303.ExistingActivationKey;
-import ru.mystamps.web.validation.jsr303.FieldsMatch;
-import ru.mystamps.web.validation.jsr303.FieldsMismatch;
-import ru.mystamps.web.validation.jsr303.UniqueLogin;
+
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/controller/dto/ActivateAccountForm.java]: I removed this extra blank line in eede8a23da7bc3bec8cd9e7d8003eb2080749269 commit.

File: src/main/java/ru/mystamps/web/support/beanvalidation/ExistingActivationKey.java [hunk 6]
```diff
@@ -34,7 +34,8 @@
 @Constraint(validatedBy = ExistingActivationKeyValidator.class)
 @Documented
 public @interface ExistingActivationKey {
-	String message() default "{ru.mystamps.web.validation.jsr303.ExistingActivationKey.message}";
+        // CheckStyle: ignore LineLength for next 1 line
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/support/beanvalidation/ExistingActivationKey.java]: I fixed indentation in 348d63779a3a61bcc6fb850eb96ca4d5a40b723b commit.

**Comment by php-coder** [comment]
Merged in e19367e73aac6f2c185d930bd9b3a7f95158e1bc and ba8271805a464faf5121644b0c7dd0c30ccb093b commits.

**Comment by GiBg1aN** [comment]
Thank you for your time and for your tips, this was my first PR

**Comment by php-coder** [comment]
@GiBg1aN I've just added you to our contributor list: https://github.com/php-coder/mystamps/wiki/team-members

Do you want to do more fixes/improvements and join our team?

**Comment by GiBg1aN** [comment]
Thank you for your proposal, maybe in the future ;)

Can I delete this branch? I don't know why but it says that there are unmerged commits

**Comment by php-coder** [comment]
> Thank you for your proposal, maybe in the future ;)

Ok

> Can I delete this branch?

Yes, sure.

> I don't know why but it says that there are unmerged commits

Yes, that because I didn't merge the commits but cherry-picked them (and also modified their descriptions).

**Comment by GiBg1aN** [comment]
Thanky you another day, maybe in the next days I'll take a look at your other issues and try to fix them.

**Comment by php-coder** [comment]
@GiBg1aN BTW how did you find this project?

**Comment by GiBg1aN** [comment]
I was looking for Java repos that had an issue labeled as "trivial"

**Pull request closed without merging**